sql/lenkwerk/storage/sql/bagger_lw_storage.sql

```diff
@@ -0,0 +1,335 @@
+
+create schema if not exists storage;
+
+
+create table storage.postgres_instance (
+   id serial not null unique,
+   host varchar,
+   port int,
+   username varchar not null,
+   status smallint not null default 0,
+   primary key (host, port),
+   check (status >= 0 and status <= 3) -- if you change this, change docs below
+);
+
+SELECT pg_catalog.pg_extension_config_dump('storage.postgres_instance', '');
+SELECT pg_catalog.pg_extension_config_dump('storage.postgres_instance_id_seq', '');
+
+COMMENT ON TABLE storage.postgres_instance IS
+$$The postgres_instance table provides enough information for Bagger components 
+to connect to the relevant database instances.  Note that authentication is NOT
+in scope and must be set separately.  We recommend certificate authentication
+or gssapi.  As a fallback, the .pgpass can be used though rotating passwords is
+not easy.
+
+DO NOT STORE PASSWORDS IN PLAIN TEXT IN THE DATABASE$$; 
+
+COMMENT ON COLUMN storage.postgres_instance.status IS 
+$$ The status here is a bit string for various statuses.  
+Currently the bits are: 
+0 - Can read 
+1 - Can write
+
+This leads to the following values:
+0 - Offline
+1 - read-only
+2 - write-only
+3 - Online (can read and write)
+
+New states may be added in the future.
+$$;
+
+create table storage.servermaps (
+   generation serial primary key,
+   server_map json not null
+);
+
+SELECT pg_catalog.pg_extension_config_dump('storage.servermaps', '');
+SELECT pg_catalog.pg_extension_config_dump('storage.servermaps_generation_seq', '');
+
+comment on table storage.servermaps is
+$$ The servermaps table stores the server maps by generation so that
+we can determine which servers are supposed to receive writes together.$$;
+
+CREATE TABLE storage.dimensions (
+   id serial NOT NULL UNIQUE,
+   ordinality INT NOT NULL UNIQUE DEFERRABLE INITIALLY DEFERRED,
+   default_val varchar not null,
+   fieldname varchar PRIMARY KEY
+);
+
+SELECT pg_catalog.pg_extension_config_dump('storage.dimensions', '');
+SELECT pg_catalog.pg_extension_config_dump('storage.dimensions_id_seq', '');
+
+
+COMMENT ON TABLE storage.dimensions IS
+$$ This contains the partitions by which data is partitioned. Default values
+can be set using the default_val field though this is not yet supported.$$;
+
+
+create table storage.indexes (
+   id serial not null unique,
+   indexname varchar(16) default 'bagger_idx',
+   access_method varchar not null, -- extensible, enforced in tooling
+   tablespc varchar not null default 'pg_default';
```

Comment:
syntax error at `;`

Reply:
oooh ouch.  reminds me re-run tests with all db params set every time I make a change to anything.